compiling boutcore

Since #1197 compiling boutcore does not work for me. My generated boutcore.pyx has the line
```
# distutils: extra_link_args =
```
If I delete that line and `make` again, boutcore compiles successfully.

The output when it fails is
```
  Generating boutcore.pyx
find: ‘/usr/lib/python3.5/dist-packages’: No such file or directory
  Generating setup.py
  Generating helper.cxx
  Generating helper.h
  Generating resolve_enum.pxd
  Generating boutcpp.pxd
Traceback (most recent call last):
  File "setup.py", line 16, in <module>
    language="c++",             # generate C++ code
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 758, in cythonize
    aliases=aliases)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 664, in create_extension_list
    kwds = deps.distutils_info(file, aliases, base).values
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 564, in distutils_info
    return (self.transitive_merge(filename, self.distutils_info0, DistutilsInfo.merge)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 574, in transitive_merge
    node, extract, merge, seen, {}, self.cimported_files)[0]
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 579, in transitive_merge_helper
    deps = extract(node)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 554, in distutils_info0
    info = self.parse_dependencies(filename)[3]
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 438, in parse_dependencies
    return parse_dependencies(source_filename)
  File "/usr/lib/python3/dist-packages/Cython/Utils.py", line 29, in wrapper
    res = cache[args] = f(*args)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 409, in parse_dependencies
    distutils_info = DistutilsInfo(source)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 198, in __init__
    value = parse_list(value)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 132, in parse_list
    if s[0] == '[' and s[-1] == ']':
IndexError: string index out of range
Makefile:19: recipe for target 'boutcore.so' failed
make: *** [boutcore.so] Error 1
```

Not sure how to debug this further. Can you help @dschwoerer?

--- a/tools/pylib/_boutcore_build/boutcore.pyx.in
+++ b/tools/pylib/_boutcore_build/boutcore.pyx.in
@@ -44,7 +44,12 @@ cat <<EOF
 # distutils: library_dirs = $lib_dirs
 # distutils: sources = helper.cxx
 # distutils: extra_compile_args = $flags
-# distutils: extra_link_args = $ldflags
+EOF
+if test ".${ldflags}" != "."
+then
+    echo "# distutils: extra_link_args = $ldflags"
+fi
+cat <<EOF
 
 # cython: binding=True
 

